src/drive/web/modules/drive/RenameInputV2.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { connect } from 'react-redux'
+
+import { withClient } from 'cozy-client'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+
+import FilenameInput from 'drive/web/modules/filelist/FilenameInput'
+import { abortRenaming } from './renameV2'
+
+const updateFileNameQuery = async (client, file, newName) => {
+  return client
+    .collection('io.cozy.files')
+    .updateAttributes(file.id, { name: newName })
+}
+
+const RenameInput = ({ abortRenaming, file, client }) => {
+  return (
+    <FilenameInput
+      name={file.name}
+      onSubmit={async newName => {
+        try {
+          await updateFileNameQuery(client, file, newName)
+        } catch {
+          Alerter.error('alert.folder_name', { folderName: name })
+        } finally {
+          abortRenaming()
+        }
+      }}
+      onAbort={abortRenaming}
+    />
+  )
+}
+
+const mapDispatchToProps = dispatch => ({
+  abortRenaming: () => dispatch(abortRenaming())
+})
+
+export default connect(
+  null,
+  mapDispatchToProps
+)(withClient(RenameInput))
```

Comment:
Nit: useClient could be used to have one less level of nesting. At the end of the migration, the withClient to useClient codemod could be used.